python/ray/util/state/state_cli.py

```diff
@@ -81,7 +81,12 @@ def _parse_filter(filter: str) -> Tuple[str, PredicateType, SupportedFilterType]
         filter[predicate_index[0] : predicate_index[1]],
         filter[predicate_index[1] :],
     )
-
+    if not key.isalpha():
+        raise ValueError(
+            f"The format of a given filter {filter} is invalid: "
+            f"key {key} is not an alphabetic string. "
+            "Please provide key=val or key!=value"
+        )
```

Comment:
I don't think we need this check since valid keys are whitelisted.

Reply:
Ya i think it is overkill moreso just added bc of huaiwei's example of ```ray list tasks --filter={state=running}``` giving invalid key error vs. invalid formatting, but should be fine with the extra line in the other message, will remove